### PR TITLE
feat: fix font size unit as point + some code style modifications

### PR DESCRIPTION
**Pull request by elysoly**
Previously we calculated the font size as a percentage of the video height.
We still do it if the font size value is not set (in point unit)

### CODE REVIEW

File: vikit/postprocessing/video_subtitle_renderer.py [hunk 1]
```diff
@@ -14,29 +14,30 @@
...
     A class to implement adding subtitle to a video
+    - font_size in point (1 px = 0.75 pt; 1 pt = 1.333333 px), if none, it will be calculated based on the video height
+
     """
 
     def __init__(
         self,
-        font_path="../../medias/arial.ttf",
-        font_size_ratio=0.045,
+        font_path=f"{os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))}/medias/arial.ttf",
```
Comment (nwettstein) [inline vikit/postprocessing/video_subtitle_renderer.py:33]: wow, I'm confused :-) what is this doing?

File: vikit/postprocessing/video_subtitle_renderer.py [hunk 2]
```diff
@@ -14,29 +14,30 @@
...
+        font_size=None,
         margin_bottom_ratio=0.04,
         margin_right_ratio=0.05,
         margin_left_ratio=0.05,
     ) -> None:
         self.font_path = font_path
-        self.font_size_ratio = font_size_ratio
+        self.font_size = font_size
```
Comment (nwettstein) [inline vikit/postprocessing/video_subtitle_renderer.py:40]: we've started adopting the practice of including the unit in the variable name, so `font_size_pt` if you don't mind

File: vikit/postprocessing/video_subtitle_renderer.py [hunk 3]
```diff
@@ -64,15 +65,15 @@ def wrap_text(self, text, max_width, font_path, font_size):
         for word in text.split():
             test_line = f"{current_line} {word}".strip()
             # Use getbbox to calculate text width
-            test_width = font.getbbox(test_line)[2]  # getbbox returns (x_min, y_min, x_max, y_max)
+            test_width = font.getbbox(test_line)[2]
```
Comment (nwettstein) [inline vikit/postprocessing/video_subtitle_renderer.py:68]: another common practice is to use unpacking with underscores in this scenario. but up to you what you find more readable:

`_, _, test_width, _ = font.getbbox(test_line)`

File: vikit/postprocessing/video_subtitle_renderer.py [hunk 4]
```diff
@@ -102,10 +103,10 @@ def add_subtitles_to_video(
         # Load the video
         logger.debug(f"Loading video from {input_video_path} to add subtitle ...")
         video = VideoFileClip(input_video_path)
-        video_duration = video.duration  # Get the original video duration
+        video_duration = video.duration
```
Comment (nwettstein) [inline vikit/postprocessing/video_subtitle_renderer.py:106]: do we even need this variable?

File: vikit/postprocessing/video_subtitle_renderer.py [hunk 5]
```diff
@@ -102,10 +103,10 @@ def add_subtitles_to_video(
...
 
         # Calculate the parameters relative to the video height
-        font_size = int(video.h * self.font_size_ratio)
+        font_size = int(video.h * 0.045) if self.font_size is None else self.font_size
```
Comment (nwettstein) [inline vikit/postprocessing/video_subtitle_renderer.py:109]: 2 points:

1. a slightly more readable idiom that's equivalent

    `font_size = self.font_size or int(video.h * 0.045)`

2. Please put 0.045 into a constant at the top of the file

File: tests/test_postprocessing.py [hunk 6]
```diff
@@ -28,9 +28,15 @@ class TestPostProcessing:
     def test_render_subtitle(self):
         with WorkingFolderContext():
 
-            subtitle_writer = VideoSubtitleRenderer()
+            subtitle_writer = VideoSubtitleRenderer(font_size=15)
```
Comment (nwettstein) [inline tests/test_postprocessing.py:31]: Should we add a test where font_size = None?

**Review by nwettstein — CHANGES_REQUESTED** [review]

**Comment by nwettstein** [comment]
Thank you by the way for doing some code cleanups along the way!!

File: vikit/postprocessing/video_subtitle_renderer.py (hunk 1 above)
Comment (elysoly) [reply vikit/postprocessing/video_subtitle_renderer.py:33]: I think once we discussed with JF, if you run tests from different directories the relative address could create problems, so it was proposed to get the current file address (os.path.abspath(__file__) ) and move upward from there instead of ../../ 
but we can change it back if you think it is not a proper way of doing it

**Review by nwettstein — APPROVED** [review]

File: vikit/postprocessing/video_subtitle_renderer.py (hunk 1 above)
Comment (nwettstein) [reply vikit/postprocessing/video_subtitle_renderer.py:35]: I think I found a nicer way. What do you think about this:

```
from pathlib import Path

font_path=str(Path(__file__).parent.parent / "medias" / "arial.ttf")
```

I haven't tested it but I think it should work. More details in https://docs.python.org/3/library/pathlib.html

Comment (elysoly) [reply vikit/postprocessing/video_subtitle_renderer.py]: yeah perfect!

**Review by Ndoradoumngue — APPROVED** [review]

File: vikit/postprocessing/video_subtitle_renderer.py [hunk 7]
```diff
@@ -131,8 +133,8 @@ def add_subtitles_to_video(
             # Sometimes srt file is longer than the real video, here is to avoid having black extra frames at the end
             if start_time < 0:
                 start_time = 0
-            if end_time > video_duration:
-                end_time = video_duration
+            if end_time > video.duration:
```
Comment (nwettstein) [inline vikit/postprocessing/video_subtitle_renderer.py:136]: minor suggestion, don't know if you like this style:

```
start_time = max(0, start_time)
end_time = min(video_duration, end_time)
```

**Review by nwettstein — APPROVED** [review]
